mocha-multi.js

```diff
@@ -35,8 +35,18 @@ process.exit = function(code) {
   process.on('exit', quit);
 }
 
-function MochaMulti(runner) {
-  var streams = initReportersAndStreams(runner, parseSetup());
+function MochaMulti(runner, options) {
+  var setup=[];
+  if( options && options.reporterOptions ) {
+    debug("options %j", options.reporterOptions);
+    var reporters=Object.keys(options.reporterOptions);
+    reporters.forEach(function(reporter) {
+      debug("%j %j", options.reporterOptions, reporter);
+      setup.push([ reporter, options.reporterOptions[reporter].stdout, options.reporterOptions[reporter].options ]);
+    });
+  }
+  debug("setup %j", setup);
+  var streams = initReportersAndStreams(runner, setup || parseSetup());
```

Comment:
What versions of Node do you test on? I can install older versions and
verify my change works

On Thu, Apr 16, 2015 at 10:45 AM, Glen Mailer <notifications@github.com>
wrote:

> In mocha-multi.js
> <https://github.com/glenjamin/mocha-multi/pull/9#discussion_r28532531>:
>
> > @@ -35,8 +35,18 @@ process.exit = function(code) {
> >    process.on('exit', quit);
> >  }
> >
> > -function MochaMulti(runner) {
> > -  var streams = initReportersAndStreams(runner, parseSetup());
> > +function MochaMulti(runner, options) {
> > +  var setup=[];
> > +  if( options && options.reporterOptions ) {
> > +    debug("options %j", options.reporterOptions);
> > +    var reporters=Object.keys(options.reporterOptions);
> > +    reporters.forEach(function(reporter) {
> > +      debug("%j %j", options.reporterOptions, reporter);
> > +      setup.push([ reporter, options.reporterOptions[reporter].stdout, options.reporterOptions[reporter].options ]);
> > +    });
> > +  }
> > +  debug("setup %j", setup);
> > +  var streams = initReportersAndStreams(runner, setup || parseSetup());
>
> Using this PR is fine, looks to be failing on Node 0.10 at the mo though
> … <#14cc35610592c7a1_>
>  On 16 Apr 2015, at 18:36, coyotebringsfire <notifications@github.com>
> wrote: In mocha-multi.js: > @@ -35,8 +35,18 @@ process.exit =
> function(code) { > process.on('exit', quit); > } > > -function
> MochaMulti(runner) { > - var streams = initReportersAndStreams(runner,
> parseSetup()); > +function MochaMulti(runner, options) { > + var setup=[];
> > + if( options && options.reporterOptions ) { > + debug("options %j",
> options.reporterOptions); > + var
> reporters=Object.keys(options.reporterOptions); > +
> reporters.forEach(function(reporter) { > + debug("%j %j",
> options.reporterOptions, reporter); > + setup.push([ reporter,
> options.reporterOptions[reporter].stdout,
> options.reporterOptions[reporter].options ]); > + }); > + } > +
> debug("setup %j", setup); > + var streams = initReportersAndStreams(runner,
> setup || parseSetup()); You're right, thanks. I made a fix and the unit
> tests pass now. I've pushed my changes to my fork, would you like a new PR?
> … — Reply to this email directly or view it on GitHub.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/glenjamin/mocha-multi/pull/9/files#r28532531>.
>



-- 
We should look for someone to eat and drink with before looking for
something to eat and drink... ~Epicurus